src/core/browser/qgsdirectoryitem.cpp

```diff
@@ -454,7 +454,9 @@ bool QgsDirectoryItem::pathShouldByMonitoredByDefault( const QString &path )
 
   // else if we know that the path is on a slow device, we don't monitor by default
   // as this can be very expensive and slow down QGIS
-  if ( QgsFileUtils::pathIsSlowDevice( path ) )
+  // Add trailing slash or windows API functions like GetDriveTypeW won't identify
+  // UNC network drives correctly
+  if ( QgsFileUtils::pathIsSlowDevice( path.endsWith( '/' ) ? path : path + '/' ) )
```

Comment:
wouldn't it make more sense to push the addition of this trailing slash into pathIsSlowDevice(), or perhaps even QgsFileUtils::driveType() itself?

Reply:
> wouldn't it make more sense to push the addition of this trailing slash into pathIsSlowDevice(), or perhaps even QgsFileUtils::driveType() itself?

That was my first implementation but it was wrong because pathIsSlowDevice (and driveType) both accept paths to files, so the correct implementations is to add the trailing slash here where we are sure it is a directory path.